### PR TITLE
Fix login issues + UI stabilization

### PR DESCRIPTION
**Pull request by brionmario**
## Purpose

1. Users with no login permission were caught in a loop when trying to login to the portal. This PR introduces an error page with a logout button so that they can login with a different account.
2. Users without the `internal_humantask_view` permission are now restricted from accessing the operations page.
3. Javascript errors should not break the UI. This PR introduces an `Error Boundary` component which can be used to wrap the UI sections for error handling.
4. When there are no discoverable applications, the applications side panel item is now hidden.

### UI Changes

**Error Boundary**

Screen Shot 2019-12-03 at 1 55 20 PM

**When there are no applications (only shown on manual navigation)**

Screen Shot 2019-12-02 at 6 18 43 PM

**When a user with no login permission tries to login to the user portal**

Screen Shot 2019-12-03 at 1 56 08 PM

**Redesigned 404 error page**

Screen Shot 2019-12-03 at 1 55 51 PM

**Consent widget icon modifications**

Screen Shot 2019-12-03 at 1 59 14 PM

### CODE REVIEW

File: apps/user-portal/src/locales/ta/views.ts [hunk 1]
```diff
@@ -144,6 +144,17 @@ export const views: Views = {
                     }
                 }
             },
+            placeholders: {
```
Comment (brionmario) [inline apps/user-portal/src/locales/ta/views.ts]: @thivi Please validate these.

File: apps/user-portal/src/locales/ta/views.ts [hunk 2]
```diff
@@ -1010,13 +1010,18 @@ export const views: Views = {
...
             subTitle: "உங்களின் கணக்கின் பாதுகாப்பை உறுதிப்படுத்த அமைப்புக்களை புதுப்பிக",
             title: "பாதுகாப்பு"
-        },
-        404: {
-            subTitle: "நீங்கள் தேடி வந்த பக்கம் இங்கில்லை.",
-            title: "நீங்கள் வழி தவறி வந்து விட்டதாக தெரிகிறது :("
         }
     },
     placeholders: {
+        404: {
```
Comment (brionmario) [inline apps/user-portal/src/locales/ta/views.ts]: @thivi Please validate this 404 block as well

**Comment by jeradrutnam** [comment]
Let's change back "404" to "Page not found". And tweak the icon and message little bit. Because icon gives more of an under construction feeling. :)

File: apps/user-portal/src/locales/ta/views.ts (hunk 2 above)
Comment (thivi) [reply apps/user-portal/src/locales/ta/views.ts]: பக்கத்தின் முகவரியினை சரி பாருங்கள் அல்லது கீழிருக்கும் பொத்தானை அழுத்தி

File: apps/user-portal/src/locales/ta/views.ts (hunk 1 above)
Comment (brionmario) [reply apps/user-portal/src/locales/ta/views.ts]: Changed. Thanks.

**Comment by brionmario** [comment]
> Let's change back "404" to "Page not found". And tweak the icon and message little bit. Because icon gives more of an under construction feeling. :)

The suggested changes have been implemented.

File: apps/user-portal/src/locales/en/views.ts [hunk 3]
```diff
@@ -957,20 +968,41 @@ export const views: Views = {
...
-        },
+        }
     },
     placeholders: {
+        404: {
+            action: "Back to home",
+            subtitles: {
+                0: "We couldn't find the page you are looking for.",
+                1: "Please check the URL or click on the button below",
+                2: "to be redirected back to the home page."
```
Comment (jeradrutnam) [inline apps/user-portal/src/locales/en/views.ts]: Remove this force break and handle it using a `max-width` controlled UI element

**Comment by brionmario** [comment]
Resolves #131, #152

**Pull request merged**